### PR TITLE
The Hissening, ports X speech variations to lizards

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/50763

## About The Pull Request
This PR adds lizard speech variation to the letter "X". Phonetically similar cases that use X as "Zee" like in Xenomorph will remain the same, Phonetically different cases that use X as "ecks" will now appear as ecks

## Why It's Good For The Game
Lizard speech only being special in their egregious use of "ssss" is boring, I think we should make them further ~~unintelligble~~ interesting with more speech variations.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

lizard hiss

</details>

## Changelog
:cl: Zytolg, RKz
add: Lizard speech now has a variation on the use of X in normal conversation. Things like "Xenomorph" will sound the same as before as it is phonetically similiar, however uses of "x-ray" will become "ecks-ray". 
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
Oh man imagine trying to spell out codewords as a lizard tot after this

File: code/modules/surgery/organs/tongue.dm [hunk 1]
```diff
@@ -69,10 +69,18 @@
 /obj/item/organ/tongue/lizard/handle_speech(datum/source, list/speech_args)
 	var/static/regex/lizard_hiss = new("s+", "g")
 	var/static/regex/lizard_hiSS = new("S+", "g")
+	var/static/regex/lizard_kss = new(@"(\w)x", "g")
+	var/static/regex/lizard_kSS = new(@"(\w)X", "g")
+	var/static/regex/lizard_ecks = new(@"\bx([\-|r|R]|\b)", "g")
+	var/static/regex/lizard_eckS = new(@"\bX([\-|r|R]|\b)", "g")
```
Comment (itsmeow) [inline code/modules/surgery/organs/tongue.dm]: `|` is not needed within regex character sets, this only applies capture groups, instead it's just capturing `|` characters twice. You also do not need to escape `-`. `[-rR]` should work fine.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/tongue.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/surgery/organs/tongue.dm]: i believe the original individual who coded it said they felt neurotic about the letters not capturing capitalized cases correctly. I will test the functionality without them

Comment (Tsar-Salat) [reply code/modules/surgery/organs/tongue.dm]: image
addressed

**Review by itsmeow — APPROVED** [review]

**Pull request merged**